### PR TITLE
feat: RHICOMPL-2450 implemented OpenSCAP result obfuscation

### PR DESCRIPTION
**Pull request by skateman**
Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:
As compliance relies on the external OpenSCAP tool, it was ignoring the `--obfuscate` and `--obfuscate-hostname` parameters. This proposed solution goes through the OpenSCAP result XML file and removes the sensitive information based on the predefined xpaths stored in a YAML file.

### CODE REVIEW

**Review by romanblanco — APPROVED** [review]
Can't spot anything wrong. Overall changes make sense :+1:

**Review by marleystipich2 — APPROVED** [review]
These changes look good to me too! I think this is a good solution. Thanks, @skateman!

**Comment by xiangce** [comment]
@skateman - This PR is still in `Draft` state, please change it to "Ready" by clicking the "Ready for review" button when it's ready. then we can move it forward. Thanks.

**Comment by skateman** [comment]
@xiangce there was some uncertainty about obfuscating MAC addresses alongside with IPs, so I had to check with our PM first. Additionally, the obfuscation is somehow ignored in the inventory side and even though our compliance report no longer contains any IP addresses or hostname. 

I stopped the report parsing on our side (compliance) so it's definitely something in the ingress-puptoo-inventory triangle that sets the hostname, but I guess the client is somehow sending it even if the obfuscation is turned on. However, I don't really know the codebase and where to look for debugging points. Can you maybe help me?

**Comment by skateman** [comment]
After some debugging, I think I found the problem. The filenames in the archive structure contain the hostname:
```
drwx------ root/root         0 2022-03-03 08:41 ./
drwx------ root/root         0 2022-03-03 08:41 ./insights-rhel7-7-insights-client.virbr0.skate-laptop-20220303074113/
-rw-r--r-- root/root   9208580 2022-03-03 08:41 ./insights-rhel7-7-insights-client.virbr0.skate-laptop-20220303074113/oscap_results-xccdf_org.ssgproject.content_profile_hipaa.xml
```

So I guess I know how to fix that :see_no_evil:

**Comment by skateman** [comment]
Apparently getting rid of the host name from the files in the archive does not help much :disappointed: somehow the upload still contains the hostname info.

**Comment by xiangce** [comment]
@skateman - Thanks for the update. The PR looks okay now. Once it's get approved by @Glutexo, we'll merge it.

@Glutexo - would you please take look at this change? Thanks.

File: insights/compliance_obfuscations.yaml
Comment (skateman) [reply insights/compliance_obfuscations.yaml:4]: Yeah, the original ticket was mentioning MAC addresses, but I guess they're not necessary. Will fix it!

File: insights/compliance_obfuscations.yaml
Comment (skateman) [reply insights/compliance_obfuscations.yaml:7]: I was working mainly with [this](https://github.com/RedHatInsights/compliance-backend/blob/master/test/fixtures/files/arf_report.xml) file but also with generated real reports.

File: insights/compliance_obfuscations.yaml
Comment (skateman) [reply insights/compliance_obfuscations.yaml:13]: See above

File: insights/compliance_obfuscations.yaml
Comment (skateman) [reply insights/compliance_obfuscations.yaml:1]: I'd prefer to cover all the possible xpaths to get rid of the IP/hostnames.

**Comment by Glutexo** [comment]
https://csrc.nist.gov/CSRC/media/Publications/nistir/7275/rev-4/final/documents/nistir-7275r4_updated-march-2012_clean.pdf

File: insights/compliance_obfuscations.yaml
Comment (Glutexo) [reply insights/compliance_obfuscations.yaml:1]: https://csrc.nist.gov/CSRC/media/Publications/nistir/7275/rev-4/final/documents/nistir-7275r4_updated-march-2012_clean.pdf

**Comment by xiangce** [comment]
@Glutexo - I think @skateman has updated it according to your feedback, May I get your approval before merging this PR? Thanks.

**Comment by Glutexo** [comment]
Looking into this now.

**Comment by clarabez** [comment]
LGTM!

**Review by Glutexo — APPROVED** [review]
It works, so let’s merge it. I may come later with some ideas because I like this work around obfuscation.

**Pull request merged**
